internal/xcrd/crd.go

```diff
@@ -118,6 +119,9 @@ func ForCompositeResourceClaim(xrd *v1.CompositeResourceDefinition) (*extv1.Cust
 		}
 		crdv.AdditionalPrinterColumns = append(crdv.AdditionalPrinterColumns, CompositeResourceClaimPrinterColumns()...)
 		for k, v := range CompositeResourceClaimSpecProps() {
+			if k == "compositeDeletePolicy" && xrd.Spec.DefaultCompositeDeletePolicy != nil {
+				v.Default = &extv1.JSON{Raw: []byte(fmt.Sprintf("\"%s\"", *xrd.Spec.DefaultCompositeDeletePolicy))}
+			}
```

Comment:
Would it make sense to pass in this default as an argument to `CompositeResourceClaimSpecProps`? I'm not really worried about performance here, but it seems a bit wasteful to do this check in the loop if we know the key will always be in the schema.

Reply:
`CompositeResourceClaimSpecProps()` is used in other places, like in `APIDryRunCompositeConfigurator`, where such parameter does not make much sense. How about that we set the default outside of the loop, something like this:

```golang
props := CompositeResourceClaimSpecProps()
if xrd.Spec.DefaultCompositeDeletePolicy != nil {
 cdp := props["compositeDeletePolicy"]
 cdp.Default = = &extv1.JSON{Raw: []byte(fmt.Sprintf("\"%s\"", *xrd.Spec.DefaultCompositeDeletePolicy))}
 props["compositeDeletePolicy"] = cdp 
}
for k, v := range props {
  crdv.Schema.OpenAPIV3Schema.Properties["spec"].Properties[k] = v
}
```

Reply:
Yeah that seems like an improvement.

I could see also adding this as variadic options to `CompositeResourceClaimSpecProps`, e.g. `CompositeResourceClaimSpecProps(WithDefaultCompositeDeletePolicy(...))` but I can see how that might be overkill.

Reply:
fixed.